torch/cuda/streams.py

```diff
@@ -119,90 +102,75 @@ def __repr__(self):
                 .format(self.device, self.cuda_stream))
 
 
-class EventHandle(ctypes.Structure):
-    IPC_HANDLE_SIZE = 64
-    _fields_ = [('reserved', ctypes.c_char * IPC_HANDLE_SIZE)]
-
-
-class Event(object):
-    r"""Wrapper around CUDA event.
+class Event(torch._C._CudaEventBase):
+    r"""Wrapper around CUDA event. Every event is associated with a device index, which
+    is the device where the event is lazily created when first recorded. However, if
+    constructed from a handle or ipc_handle() is called before it is ever recorded, the
+    device index will be the sames as the device of ``torch.cuda.current_stream()``.
 
     Arguments:
-        enable_timing (bool): indicates if the event should measure time
+        enable_timing (bool, optional): indicates if the event should measure time
             (default: ``False``)
-        blocking (bool): if ``True``, :meth:`wait` will be blocking (default: ``False``)
+        blocking (bool, optional): if ``True``, :meth:`wait` will be blocking (default: ``False``)
```

Comment:
This predates your change, but the description is wrong. `cudaEventBlockingSync` affects the type of synchronization used in `synchronize()` (busy-wait vs. blocking). It doesn't affect `wait()`.